_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,356 @@
+---
+id: d0f69e
+name: All table header cells have assigned data cells
+rule_type: atomic
+description: |
+  This rule checks that each table header has assigned data cells in a table element.
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+acknowledgements:
+  authors:
+    - Jey Nandakumar
+    - Audrey Maniez
+---
+
+## Applicability
+
+The rule applies to any HTML element that
+
+- has the [semantic role][] of [rowheader][] or [columnheader][] and;
+- is [visible][] and;
```

Comment:
I'm wondering why visibility is important in this proposed rule. 

In "Inapplicable 7," below, we have a passing example which is marked inapplicable because of this "visibility" clause. If the table were not passing but also not visible (due to positioning), would it still be inapplicable? 

A screen reader would still be able to navigate the non-visible table and, if the table were non-passing, it would encounter what is the stated purpose of this rule: to ensure that all headers have assigned data cells.

It seems that the crucial aspect is inclusion in the accessibility tree, not visibility. A non-visible table can still be navigated by a screen reader as though it were visible, so long as it is in the accessibility tree.

Reply:
@kensgists Good point. I have updated accordingly. The visible criteria is not essential on a table as you pointed out that screen readers will pick this up either how. 

But from our tests, we found that visible on the applicable element, in this case the element with a role of row or column header is essential, as for examples a table marked up as below, 
does not associate headings as expected.

```
<table>
  <tr style="display:none;">
    <th>Projects</th>
    <th>Progress</th>
  </tr>
  <tr>
    <td>My Project</td>
    <td>15%</td>
  </tr>
</table>
```

**IE + JAWS**: Does not work (Reads: "My Project" "15%" as plain text)
**Chrome + JAWS**: Does not work (Reads: "My Project" "15%" as plain text)
**Firefox + NVDA**: does not work. NVDA picks up a table with one row and two
columns: Column 1 "My Projects", Column 2 "15%".
**Safari + VO**: Does not work